test/e2e/cluster_samples_operator_test.go

```diff
@@ -53,4 +91,157 @@ func TestClusterSamplesOperator(t *testing.T) {
 	if err != nil {
 		t.Errorf("error waiting for example imagestreamtag to appear: %v", err)
 	}
+
+	is := &imageapiv1.ImageStream{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ImageStream",
+			APIVersion: imageapiv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "jenkins",
+			Namespace: "openshift",
+		},
+	}
+
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := sdk.Get(is); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("error waiting for example imagestream to appear: %v", err)
+	}
+
+}
+
+func verifyTemplatesPresent(t *testing.T) {
+	// as the content will vary over time as openshift/library evolves
+	// sample a random set of templates related artifacts
+	templates := []string{"jenkins-ephemeral", "redis-ephemeral"}
+	template := &templatev1.Template{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Template",
+			APIVersion: templatev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift",
+		},
+	}
+
+	for _, name := range templates {
+		template.Name = name
+		err := wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+			if err := sdk.Get(template); err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			t.Errorf("error waiting for example template %s to appear: %v", name, err)
+		}
+	}
+}
+
+func TestImageStreamInOpenshiftNamespace(t *testing.T) {
+	verifyOperatorUp(t)
+	verifyImagesPresent(t)
+	verifyTemplatesPresent(t)
+}
+
+func TestRecreateSamplesResourceAfterDelete(t *testing.T) {
+	sr := verifyOperatorUp(t)
+
+	oldTime := sr.CreationTimestamp
+
+	err := sdk.Delete(sr, sdk.WithDeleteOptions(&metav1.DeleteOptions{}))
+	if err != nil {
+		t.Errorf("error deleting samplesresource %v", err)
+	}
+
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := sdk.Get(sr); err != nil {
+			return false, nil
+		}
+		if sr.CreationTimestamp == oldTime {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("creation times the same after delete: %v, %v, %#v", oldTime, sr.CreationTimestamp, sr)
+	}
+
+	err = verifyConditionsCompleteSamplesAdded(sr)
+	if err != nil {
+		t.Errorf("samples not re-established after delete %#v", sr)
+	}
+
+	verifyImagesPresent(t)
+	verifyTemplatesPresent(t)
```

Comment:
should confirm the templates+imagestreams were also recreated since they should have been removed when you deleted the samples resource.

Reply:
Given the recreate is immediate in the delete case this proved difficult to do

Since the function which removes the samples is the same one used in the managementState: Removed scenario, I'm deferring revalidating here

Reply:
creation TS compare part of commit I just pushed